homeassistant/components/homekit/__init__.py

```diff
@@ -455,40 +467,44 @@ class HomeKit:
 
     def __init__(
         self,
-        hass,
-        name,
-        port,
-        ip_address,
-        entity_filter,
-        exclude_accessory_mode,
-        entity_config,
-        homekit_mode,
-        advertise_ip=None,
-        entry_id=None,
-        entry_title=None,
-        devices=None,
-    ):
+        hass: HomeAssistant,
+        name: str,
+        port: int,
+        ip_address: str | None,
+        entity_filter: EntityFilter,
+        exclude_accessory_mode: bool,
+        entity_config: dict,
+        homekit_mode: str,
+        advertise_ip: str | None = None,
+        entry_id: str | None = None,
+        entry_title: str | None = None,
+        devices: Iterable[str] | None = None,
+    ) -> None:
         """Initialize a HomeKit object."""
+        assert entry_title is not None
+        assert entry_id is not None
```

Comment:
Uh?

Why are they typed/defaulted as `None` in that case?

Reply:
They shouldn't ever be `None` but it will break many of tests if I change it in this PR as the tests rely on order and they were previously defaulting to `None`
```
        advertise_ip=None,
        entry_id=None,
        entry_title=None,
        devices=None,
```

Reply:
Scratch that, I think I already fixed the affected tests.